packages/editor/src/client/routes/commentList.tsx

```diff
@@ -0,0 +1,501 @@
+                switch (confirmAction) {
+                  case ConfirmAction.Approve:
+                    await approveComment({
+                      variables: {
+                        id: currentComment.id
+                      },
+                      update: cache => {
+                        const query = cache.readQuery<CommentListQuery>({
+                          query: CommentListDocument,
+                          variables: commentListVariables
+                        })
+
+                        if (!query) return
+
+                        cache.writeQuery<CommentListQuery>({
+                          query: CommentListDocument,
+                          data: {
+                            comments: {
+                              ...query.comments,
+                              nodes: query.comments.nodes.filter(
+                                comment => comment.id !== currentComment.id
+                              )
+                            }
+                          },
+                          variables: commentListVariables
+                        })
+                      }
+                    })
+                    setConfirmationDialogOpen(false)
+                    resetCurrentCommentState()
+                    break
+                  case ConfirmAction.RequestChanges:
+                    if (!rejectionReason) return
+                    await requestChanges({
+                      variables: {
+                        id: currentComment.id,
+                        rejectionReason
+                      },
+                      update: cache => {
+                        const query = cache.readQuery<CommentListQuery>({
+                          query: CommentListDocument,
+                          variables: commentListVariables
+                        })
+
+                        if (!query) return
+
+                        cache.writeQuery<CommentListQuery>({
+                          query: CommentListDocument,
+                          data: {
+                            comments: {
+                              ...query.comments,
+                              nodes: query.comments.nodes.filter(
+                                comment => comment.id !== currentComment.id
+                              )
+                            }
+                          },
+                          variables: commentListVariables
+                        })
+                      }
+                    })
+                    setConfirmationDialogOpen(false)
+                    resetCurrentCommentState()
+                    break
+                  case ConfirmAction.Reject:
+                    if (!rejectionReason) return
+                    await rejectComment({
+                      variables: {
+                        id: currentComment.id,
+                        rejectionReason
+                      },
+                      update: cache => {
```

Comment:
Try not to repeat yourself. The update function seems to always do exactly the same.  I think you can move that into a function.